feat:add setting min base fee guide (#1849)
ethereum-optimism/docs: chain-operators/guides/features/setting-min-base-fee.mdx

---
title: Set the Minimum Base Fee
description: Learn how to set a minimum base fee on your OP-Stack Chain
---

## Overview

> | Parameter | Type | Default | Recommended | Max Allowed for Standard Chains | Units |

Reviewer: I see the max allowed for std chains being 1_000_000_000
- https://specs.optimism.io/protocol/configurability.html#minimum-base-fee
- https://github.com/ethereum-optimism/superchain-registry/blob/main/validation/standard/standard-config-params-mainnet.toml

Reviewer: might be a good idea to link the specs for this restriction

Author: I think you're looking at the resource config max for deposit transactions. The value I got was from this PR into the registry from @geoknee https://github.com/ethereum-optimism/superchain-registry/pull/1127

Reviewer: Okay that looks good

Reviewer 2: Yes that resource config is a constant source of confusion. Value lgtm.

| ---------- | ---- | -------- | ------------ | ------------ | ------ |
| <code>minBaseFee</code> | Number | 0 (disabled) | 100,000 | 10,000,000,000 | wei |

The <strong>Minimum Base Fee</strong> (`minBaseFee`) configuration was introduced in the <strong>Jovian</strong> release of the OP Stack. 
This feature allows chain operators to specify a minimum L2 base fee to maintain a predictable minimum transaction costs and avoids excessively long priority fee auctions that can occur when the base fee falls too low.

One situation where the minimum base fee may kick in, is when the [batcher is throttled](https://docs.optimism.io/chain-operators/guides/configuration/batcher#batcher-sequencer-throttling) for long enough. If the minBaseFee isn't enabled, the base fee can drop all the down to 1 wei, which can take awhile to recover back to a stable base fee.

The steps below explain how to update the `minBaseFee` parameter on-chain using the <code>SystemConfig</code> contract.

> <Info>

Reviewer: I'd say, make this a warning instead of an info call out

Author: done

  Setting the `minBaseFee` too high may make transactions harder to include for users.
</Info>

## How to Update the Minimum Base Fee

<Steps>
  <Step title="Find your SystemConfig contract address">
    The [SystemConfig](https://specs.optimism.io/protocol/system-config.html) contract stores configurable protocol parameters such as gas limits and fee settings.  
    You can find its proxy address in the [state.json generated by op-deployer](https://docs.optimism.io/chain-operators/tutorials/create-l2-rollup/op-deployer-setup#deploy-l1-contracts).
  </Step>

  <Step title="Call the setMinimumBaseFee() method">
    To update the minimum base fee, call the following method on the `SystemConfig` contract:

    `setMinBaseFee(uint64 minBaseFee) external onlyOwner;`

>     <Info>

Reviewer: I think you should bump this note to right under the H2

Author: done

    Note that the owner of the `SystemConfig` contract is the [System Config Owner address](https://docs.optimism.io/chain-operators/reference/privileged-roles#system-config-owner), so this transaction must be sent from that address.
    </Info>
    Example (using [cast](https://getfoundry.sh/cast/reference/cast/)):

    ```
    cast send <SYSTEM_CONFIG_PROXY_ADDRESS> "setMinBaseFee(uint64)" 100000 
    ```
  </Step>

  <Step title="Verify the new value">
    After the transaction confirms, verify the current value by calling the following getter method on the `SystemConfig` contract:
    
    `function minBaseFee() external view returns (uint64);`

    Example (using [cast](https://getfoundry.sh/cast/reference/cast/)):

    ```
    cast call <SYSTEM_CONFIG_PROXY_ADDRESS> "minBaseFee()"
    ```
  </Step>
</Steps>

---

## References

* [Minimum Base Fee Configuration Spec](https://specs.optimism.io/protocol/jovian/system-config.html#minimum-base-fee-configuration)
* [Jovian Upgrade Spec](https://specs.optimism.io/protocol/jovian/overview.html)

> * [SystemConfig Contract](https://specs.optimism.io/protocol/system-config.html)

Reviewer: ```suggestion
* [SystemConfig Contract Spec](https://specs.optimism.io/protocol/system-config.html)
```

---
title: Set the Minimum Base Fee
description: Learn how to set a minimum base fee on your OP-Stack Chain
---

## Overview

| Parameter | Type | Default | Recommended | Max Allowed for Standard Chains | Units |
| ---------- | ---- | -------- | ------------ | ------------ | ------ |
| <code>minBaseFee</code> | Number | 0 (disabled) | 100,000 | 10,000,000,000 | wei |

> The <strong>Minimum Base Fee</strong> (`minBaseFee`) configuration was introduced in the <strong>Jovian</strong> release of the OP Stack. 

Reviewer 2: ```suggestion
The <strong>Minimum Base Fee</strong> (`minBaseFee`) configuration was introduced in the <strong>Jovian</strong> hardfork. 
```
Ideally we add a link to the notice page, too.

> This feature allows chain operators to specify a minimum L2 base fee to maintain a predictable minimum transaction costs and avoids excessively long priority fee auctions that can occur when the base fee falls too low.

Reviewer 2: ```suggestion
This feature allows chain operators to specify a minimum L2 base fee to which can help avoid excessively long priority fee auctions that can occur when the base fee falls too low.
```

> One situation where the minimum base fee may kick in, is when the [batcher is throttled](https://docs.optimism.io/chain-operators/guides/configuration/batcher#batcher-sequencer-throttling) for long enough. If the minBaseFee isn't enabled, the base fee can drop all the down to 1 wei, which can take awhile to recover back to a stable base fee.

Reviewer 2: ```suggestion
When [batcher-sequencer throttling is active](https://docs.optimism.io/chain-operators/guides/configuration/batcher#batcher-sequencer-throttling) for long enough and the minBaseFee isn't enabled (or is zero), the base fee can drop all the down to 1 wei. It can take a long time to recover back to a stable base fee.
```
